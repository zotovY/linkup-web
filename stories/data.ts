export const user: User = {
    name: "Yaroslav Zotov",
    id: 1,
    email: "the1ime@yandex.ru",
    createdAt: new Date(),
    password: "12345678",
    theme: 0,
    links: [],
    profileImagePath: "https://sun9-70.userapi.com/impg/Afr3SX4BeAJzLemVyNMhHhr-5qvMlj1MedV0wA/5B6s0MBC6Y8.jpg?size=1728x2160&quality=96&sign=7a48ae45b2c91959995e58331fa10926&type=album"
}

export const link: Link = {
    createdAt: new Date(),
    href: "https://github.com/zotovY",
    iconName: "logo-github",
    id: 1,
    subtitle: "This is my github account",
    title: "Github",
    user: user,
}
