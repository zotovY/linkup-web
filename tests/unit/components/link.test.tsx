/// <reference types="@types/jest" />
import { render, cleanup, fireEvent } from "../../test-utils";
import '@testing-library/jest-dom';
import 'jest-styled-components';

import LinkComponent, { Props } from "@/components/link";
import { link } from "../../data";

describe("Testing link component", () => {

    afterEach(cleanup);

    const props: Props = {
        link,
        remove: () => {
        },
        save: () => {
        },
        onChange: () => {
        },

    }

    test("should renders correctly", () => {
        const { asFragment } = render(<LinkComponent { ...props } />);
        expect(asFragment()).toMatchSnapshot();
    });

    test("should renders correctly in edit mode", () => {
        const { asFragment, getByTestId } = render(<LinkComponent { ...props } />);
        const header = getByTestId("link-header");
        fireEvent.click(header);

        expect(asFragment()).toMatchSnapshot();
    });

    test("should change title", () => {
        const expectedLink = { ...props.link, title: "HEEEEY" };
        let link: Link | undefined;
        const changeLink = (l: Link) => link = l;

        // Arrange
        const { getByTestId, getByPlaceholderText, getByText } = render(
                <LinkComponent { ...props } save={ changeLink }/>
        );

        // Act
        fireEvent.click(getByTestId("link-header"));
        fireEvent.change(getByPlaceholderText("Title"), { target: { value: expectedLink.title } });
        fireEvent.click(getByText("Save"));

        // Assert
        expect(getByText(expectedLink.title)).toBeTruthy();
        expect(link).toEqual(expectedLink);
    });

    test("should change subtitle", () => {
        const expectedLink = { ...props.link, subtitle: "XD" };
        let link: Link | undefined;
        const changeLink = (l: Link) => link = l;

        // Arrange
        const { getByTestId, getByPlaceholderText, getByText } = render(
                <LinkComponent { ...props } save={ changeLink }/>
        );

        // Act
        fireEvent.click(getByTestId("link-header"));
        fireEvent.change(getByPlaceholderText("Subtitle"), { target: { value: expectedLink.subtitle } });
        fireEvent.click(getByText("Save"));

        // Assert
        expect(getByText(expectedLink.subtitle)).toBeTruthy();
        expect(link).toEqual(expectedLink);
    });

    test("should change href", () => {
        const expectedLink = { ...props.link, href: "https://vk.com/feed" };
        let link: Link | undefined;
        const changeLink = (l: Link) => link = l;

        // Arrange
        const { getByTestId, getByPlaceholderText, getByText } = render(
                <LinkComponent { ...props } save={ changeLink }/>
        );

        // Act
        fireEvent.click(getByTestId("link-header"));
        fireEvent.change(getByPlaceholderText("Link"), { target: { value: expectedLink.href } });
        fireEvent.click(getByText("Save"));

        // Assert
        expect(link).toEqual(expectedLink);
    });

    test("should change icon", () => {
        const expectedLink = { ...props.link, iconName: "logo-instagram" };
        let link: Link | undefined;
        const changeLink = (l: Link) => link = l;

        // Arrange
        const { getByTestId, getByPlaceholderText, getByText } = render(
                <LinkComponent { ...props } save={ changeLink }/>
        );

        // Act
        fireEvent.click(getByTestId("link-header"));
        fireEvent.click(getByTestId("link-icon-selector"));
        fireEvent.change(getByPlaceholderText("Enter your icon name"), { target: { value: "Instagram" } });
        fireEvent.click(getByTestId("icon-variant"));
        fireEvent.click(getByText("Save"));

        // Assert
        expect(link).toEqual(expectedLink);
    });

    test("should delete link", () => {
        let isDeleted = false;
        const deleteLink = () => isDeleted = true;

        // Arrange
        const { getByTestId, getByText } = render(
                <LinkComponent { ...props } remove={ deleteLink }/>
        );

        // Act
        fireEvent.click(getByTestId("link-header"));
        fireEvent.click(getByText("Delete"));

        // Assert
        expect(isDeleted).toEqual(true);
    });

    test("should validate link", () => {
        const wrongData = {
            title: "",
            subtitle: "",
            href: "loook at me",
        }
        let isSaved = false;
        const onSave = () => isSaved = true;
        const { getByPlaceholderText, getByTestId, getByText } = render(<LinkComponent { ...props } save={ onSave }/>)

        // Act
        fireEvent.click(getByTestId("link-header"));
        fireEvent.change(getByPlaceholderText("Title"), { target: { value: wrongData.title } });
        fireEvent.change(getByPlaceholderText("Subtitle"), { target: { value: wrongData.subtitle } });
        fireEvent.change(getByPlaceholderText("Link"), { target: { value: wrongData.href } });
        fireEvent.click(getByText("Save"));

        // Arrange
        expect(isSaved).toBe(false);
        expect(getByTestId("link-header")).toHaveAttribute("open");
        expect(getByPlaceholderText("Title")).toHaveStyle("border: 1.5px solid #ED2E7E;")
        expect(getByPlaceholderText("Subtitle")).toHaveStyle("border: 1.5px solid #ED2E7E;")
        expect(getByPlaceholderText("Link")).toHaveStyle("border: 1.5px solid #ED2E7E;")
    });
});
