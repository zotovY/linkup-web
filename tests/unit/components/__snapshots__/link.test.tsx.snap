// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing link component should renders correctly 1`] = `
<DocumentFragment>
  .c2 svg {
  height: 36px;
  width: 36px;
  color: #5f2eea;
  fill: #5f2eea;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.c6 svg {
  height: 22px;
  width: 22px;
  color: #D9DDE9;
  fill: #D9DDE9;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.c16 svg {
  height: 42px;
  width: 42px;
  color: #5f2eea;
  fill: #5f2eea;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: calc(100% - 46px);
}

.c3 span.title {
  font-size: 15px;
  font-weight: 500;
  color: #14142b;
}

.c3 span.subtitle {
  font-size: 12px;
  color: #4e4b66;
}

.c11 {
  border: 1.5px solid #D9DDE9;
  padding: 7px 14px;
  border-radius: 10px;
  font-size: 15px;
  line-height: 28px;
  outline: none;
  background: white;
  z-index: 10;
  width: 100%;
  -webkit-transition: border 200ms;
  transition: border 200ms;
}

.c11:focus {
  border: 1.5px solid #14142B;
}

.c15 {
  background: #F3EFFF;
  border-radius: 16px;
  padding: 9px;
  width: 60px;
  height: 60px;
  cursor: pointer;
}

.c17 {
  font-size: 16px;
  line-height: 24px;
  color: #5f2eea;
  cursor: pointer;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c13 .c14 {
  margin-right: 20px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 10px;
  border-radius: 16px;
  margin-bottom: 20px;
  background-color: #FFFFFF;
  -webkit-transition: background-color 200ms ease;
  transition: background-color 200ms ease;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 69px;
  margin-bottom: 0;
  width: 100%;
  pointer-events: initial;
}

.c1:hover {
  background-color: #fEfEfE;
}

.c9 {
  width: calc(100% - 20px);
  overflow: hidden;
  height: 0;
  -webkit-transition: height 300ms ease,border-top 200ms;
  transition: height 300ms ease,border-top 200ms;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-top: none;
  margin: 0 10px;
  padding-top: 20px;
}

.c9 .c10 {
  margin-bottom: 20px;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c18 {
  font-size: 16px;
  line-height: 24px;
  color: #ed2e7e;
  cursor: pointer;
}

.c8 {
  font-size: 16px;
  line-height: 24px;
  color: #5f2eea;
  cursor: pointer;
  position: absolute;
  left: 100%;
  z-index: 100;
  pointer-events: all;
}

.c5 {
  -webkit-flex: 0 0 45px;
  -ms-flex: 0 0 45px;
  flex: 0 0 45px;
  height: 24px;
  overflow: hidden;
  position: relative;
  margin-top: 13.5px;
}

.c5 .chevron {
  position: absolute;
  right: 0;
}

.c5 .c7,
.c5 .chevron {
  -webkit-transition: -webkit-transform 200ms ease;
  -webkit-transition: transform 200ms ease;
  transition: transform 200ms ease;
}

.c5[data-open=true] .c7 {
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
}

.c5[data-open=true] .chevron {
  -webkit-transform: translateX(calc(-100% - 22px));
  -ms-transform: translateX(calc(-100% - 22px));
  transform: translateX(calc(-100% - 22px));
}

.c0 {
  height: 69px;
  -webkit-transition: height 200ms ease;
  transition: height 200ms ease;
  background-color: white;
  border-radius: 16px;
}

.c0.lang-ru .c4 {
  -webkit-flex: 0 0 81px;
  -ms-flex: 0 0 81px;
  flex: 0 0 81px;
}

.c0.lang-ru .c4[data-open=true] .chevron {
  -webkit-transform: translateX(calc(-100% - 60px));
  -ms-transform: translateX(calc(-100% - 60px));
  transform: translateX(calc(-100% - 60px));
}

<div
    class="c0 link-component lang-en"
    data-testid="link"
  >
    <div
      class="c1"
      data-testid="link-header"
    >
      <span
        class="c2"
        color="#5f2eea"
        height="36px"
        rotate="0"
        width="36px"
      >
        <svg
          class=""
          style="margin-right: 10px; margin-top: 5px;"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 32C132.3 32 32 134.9 32 261.7c0 101.5 64.2 187.5 153.2 217.9a17.56 17.56 0 003.8.4c8.3 0 11.5-6.1 11.5-11.4 0-5.5-.2-19.9-.3-39.1a102.4 102.4 0 01-22.6 2.7c-43.1 0-52.9-33.5-52.9-33.5-10.2-26.5-24.9-33.6-24.9-33.6-19.5-13.7-.1-14.1 1.4-14.1h.1c22.5 2 34.3 23.8 34.3 23.8 11.2 19.6 26.2 25.1 39.6 25.1a63 63 0 0025.6-6c2-14.8 7.8-24.9 14.2-30.7-49.7-5.8-102-25.5-102-113.5 0-25.1 8.7-45.6 23-61.6-2.3-5.8-10-29.2 2.2-60.8a18.64 18.64 0 015-.5c8.1 0 26.4 3.1 56.6 24.1a208.21 208.21 0 01112.2 0c30.2-21 48.5-24.1 56.6-24.1a18.64 18.64 0 015 .5c12.2 31.6 4.5 55 2.2 60.8 14.3 16.1 23 36.6 23 61.6 0 88.2-52.4 107.6-102.3 113.3 8 7.1 15.2 21.1 15.2 42.5 0 30.7-.3 55.5-.3 63 0 5.4 3.1 11.5 11.4 11.5a19.35 19.35 0 004-.4C415.9 449.2 480 363.1 480 261.7 480 134.9 379.7 32 256 32z"
          />
        </svg>
      </span>
      <div
        class="c3"
      >
        <span
          class="title"
        >
          Github
        </span>
        <span
          class="subtitle"
        >
          This is my github account
        </span>
      </div>
      <div
        class="c4 c5"
        data-open="false"
      >
        <span
          class="c6"
          color="#D9DDE9"
          height="22px"
          rotate="0"
          width="22px"
        >
          <svg
            class="chevron"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M184 112l144 144-144 144"
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="48"
            />
          </svg>
        </span>
        <span
          class="c7 c8"
        >
          Save
        </span>
      </div>
    </div>
    <div
      class="c9"
      data-testid="link-edit"
    >
      <input
        class="c10 c11"
        placeholder="Title"
        value="Github"
      />
      <input
        class="c10 c11"
        placeholder="Subtitle"
        value="This is my github account"
      />
      <input
        class="c10 c11"
        placeholder="Link"
        value="https://github.com/zotovY"
      />
      <div
        class="c12"
      >
        <div
          class="c13"
          data-testid="link-icon-selector"
        >
          <div
            class="c14 c15"
          >
            <span
              class="c16"
              color="#5f2eea"
              height="42px"
              rotate="0"
              width="42px"
            >
              <svg
                class=""
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 32C132.3 32 32 134.9 32 261.7c0 101.5 64.2 187.5 153.2 217.9a17.56 17.56 0 003.8.4c8.3 0 11.5-6.1 11.5-11.4 0-5.5-.2-19.9-.3-39.1a102.4 102.4 0 01-22.6 2.7c-43.1 0-52.9-33.5-52.9-33.5-10.2-26.5-24.9-33.6-24.9-33.6-19.5-13.7-.1-14.1 1.4-14.1h.1c22.5 2 34.3 23.8 34.3 23.8 11.2 19.6 26.2 25.1 39.6 25.1a63 63 0 0025.6-6c2-14.8 7.8-24.9 14.2-30.7-49.7-5.8-102-25.5-102-113.5 0-25.1 8.7-45.6 23-61.6-2.3-5.8-10-29.2 2.2-60.8a18.64 18.64 0 015-.5c8.1 0 26.4 3.1 56.6 24.1a208.21 208.21 0 01112.2 0c30.2-21 48.5-24.1 56.6-24.1a18.64 18.64 0 015 .5c12.2 31.6 4.5 55 2.2 60.8 14.3 16.1 23 36.6 23 61.6 0 88.2-52.4 107.6-102.3 113.3 8 7.1 15.2 21.1 15.2 42.5 0 30.7-.3 55.5-.3 63 0 5.4 3.1 11.5 11.4 11.5a19.35 19.35 0 004-.4C415.9 449.2 480 363.1 480 261.7 480 134.9 379.7 32 256 32z"
                />
              </svg>
            </span>
          </div>
          <span
            class="c17"
          >
            Change icon
          </span>
        </div>
        <span
          class="c18"
        >
          Delete
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Testing link component should renders correctly in edit mode 1`] = `
<DocumentFragment>
  .c2 svg {
  height: 36px;
  width: 36px;
  color: #5f2eea;
  fill: #5f2eea;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.c6 svg {
  height: 22px;
  width: 22px;
  color: #D9DDE9;
  fill: #D9DDE9;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.c16 svg {
  height: 42px;
  width: 42px;
  color: #5f2eea;
  fill: #5f2eea;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: calc(100% - 46px);
}

.c3 span.title {
  font-size: 15px;
  font-weight: 500;
  color: #14142b;
}

.c3 span.subtitle {
  font-size: 12px;
  color: #4e4b66;
}

.c11 {
  border: 1.5px solid #D9DDE9;
  padding: 7px 14px;
  border-radius: 10px;
  font-size: 15px;
  line-height: 28px;
  outline: none;
  background: white;
  z-index: 10;
  width: 100%;
  -webkit-transition: border 200ms;
  transition: border 200ms;
}

.c11:focus {
  border: 1.5px solid #14142B;
}

.c15 {
  background: #F3EFFF;
  border-radius: 16px;
  padding: 9px;
  width: 60px;
  height: 60px;
  cursor: pointer;
}

.c17 {
  font-size: 16px;
  line-height: 24px;
  color: #5f2eea;
  cursor: pointer;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c13 .c14 {
  margin-right: 20px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 10px;
  border-radius: 16px;
  margin-bottom: 20px;
  background-color: #FFFFFF;
  -webkit-transition: background-color 200ms ease;
  transition: background-color 200ms ease;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 69px;
  margin-bottom: 0;
  width: 100%;
  pointer-events: none;
}

.c1:hover {
  background-color: #fEfEfE;
}

.c20 .c10 {
  margin-bottom: 20px;
}

.c9 {
  width: calc(100% - 20px);
  overflow: hidden;
  height: calc(100% - 69px);
  -webkit-transition: height 300ms ease,border-top 200ms;
  transition: height 300ms ease,border-top 200ms;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-top: 1px solid #D9DDE9;
  margin: 0 10px;
  padding-top: 20px;
}

.c9 .c10 {
  margin-bottom: 20px;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c18 {
  font-size: 16px;
  line-height: 24px;
  color: #ed2e7e;
  cursor: pointer;
}

.c8 {
  font-size: 16px;
  line-height: 24px;
  color: #5f2eea;
  cursor: pointer;
  position: absolute;
  left: 100%;
  z-index: 100;
  pointer-events: all;
}

.c5 {
  -webkit-flex: 0 0 45px;
  -ms-flex: 0 0 45px;
  flex: 0 0 45px;
  height: 24px;
  overflow: hidden;
  position: relative;
  margin-top: 13.5px;
}

.c5 .chevron {
  position: absolute;
  right: 0;
}

.c5 .c7,
.c5 .chevron {
  -webkit-transition: -webkit-transform 200ms ease;
  -webkit-transition: transform 200ms ease;
  transition: transform 200ms ease;
}

.c5[data-open=true] .c7 {
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
}

.c5[data-open=true] .chevron {
  -webkit-transform: translateX(calc(-100% - 22px));
  -ms-transform: translateX(calc(-100% - 22px));
  transform: translateX(calc(-100% - 22px));
}

.c19.lang-ru .c4 {
  -webkit-flex: 0 0 81px;
  -ms-flex: 0 0 81px;
  flex: 0 0 81px;
}

.c19.lang-ru .c4[data-open=true] .chevron {
  -webkit-transform: translateX(calc(-100% - 60px));
  -ms-transform: translateX(calc(-100% - 60px));
  transform: translateX(calc(-100% - 60px));
}

.c0 {
  height: 355px;
  -webkit-transition: height 200ms ease;
  transition: height 200ms ease;
  background-color: white;
  border-radius: 16px;
}

.c0.lang-ru .c4 {
  -webkit-flex: 0 0 81px;
  -ms-flex: 0 0 81px;
  flex: 0 0 81px;
}

.c0.lang-ru .c4[data-open=true] .chevron {
  -webkit-transform: translateX(calc(-100% - 60px));
  -ms-transform: translateX(calc(-100% - 60px));
  transform: translateX(calc(-100% - 60px));
}

<div
    class="c0 link-component lang-en"
    data-testid="link"
  >
    <div
      class="c1"
      data-testid="link-header"
      open=""
    >
      <span
        class="c2"
        color="#5f2eea"
        height="36px"
        rotate="0"
        width="36px"
      >
        <svg
          class=""
          style="margin-right: 10px; margin-top: 5px;"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 32C132.3 32 32 134.9 32 261.7c0 101.5 64.2 187.5 153.2 217.9a17.56 17.56 0 003.8.4c8.3 0 11.5-6.1 11.5-11.4 0-5.5-.2-19.9-.3-39.1a102.4 102.4 0 01-22.6 2.7c-43.1 0-52.9-33.5-52.9-33.5-10.2-26.5-24.9-33.6-24.9-33.6-19.5-13.7-.1-14.1 1.4-14.1h.1c22.5 2 34.3 23.8 34.3 23.8 11.2 19.6 26.2 25.1 39.6 25.1a63 63 0 0025.6-6c2-14.8 7.8-24.9 14.2-30.7-49.7-5.8-102-25.5-102-113.5 0-25.1 8.7-45.6 23-61.6-2.3-5.8-10-29.2 2.2-60.8a18.64 18.64 0 015-.5c8.1 0 26.4 3.1 56.6 24.1a208.21 208.21 0 01112.2 0c30.2-21 48.5-24.1 56.6-24.1a18.64 18.64 0 015 .5c12.2 31.6 4.5 55 2.2 60.8 14.3 16.1 23 36.6 23 61.6 0 88.2-52.4 107.6-102.3 113.3 8 7.1 15.2 21.1 15.2 42.5 0 30.7-.3 55.5-.3 63 0 5.4 3.1 11.5 11.4 11.5a19.35 19.35 0 004-.4C415.9 449.2 480 363.1 480 261.7 480 134.9 379.7 32 256 32z"
          />
        </svg>
      </span>
      <div
        class="c3"
      >
        <span
          class="title"
        >
          Github
        </span>
        <span
          class="subtitle"
        >
          This is my github account
        </span>
      </div>
      <div
        class="c4 c5"
        data-open="true"
      >
        <span
          class="c6"
          color="#D9DDE9"
          height="22px"
          rotate="0"
          width="22px"
        >
          <svg
            class="chevron"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M184 112l144 144-144 144"
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="48"
            />
          </svg>
        </span>
        <span
          class="c7 c8"
        >
          Save
        </span>
      </div>
    </div>
    <div
      class="c9"
      data-testid="link-edit"
    >
      <input
        class="c10 c11"
        placeholder="Title"
        value="Github"
      />
      <input
        class="c10 c11"
        placeholder="Subtitle"
        value="This is my github account"
      />
      <input
        class="c10 c11"
        placeholder="Link"
        value="https://github.com/zotovY"
      />
      <div
        class="c12"
      >
        <div
          class="c13"
          data-testid="link-icon-selector"
        >
          <div
            class="c14 c15"
          >
            <span
              class="c16"
              color="#5f2eea"
              height="42px"
              rotate="0"
              width="42px"
            >
              <svg
                class=""
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 32C132.3 32 32 134.9 32 261.7c0 101.5 64.2 187.5 153.2 217.9a17.56 17.56 0 003.8.4c8.3 0 11.5-6.1 11.5-11.4 0-5.5-.2-19.9-.3-39.1a102.4 102.4 0 01-22.6 2.7c-43.1 0-52.9-33.5-52.9-33.5-10.2-26.5-24.9-33.6-24.9-33.6-19.5-13.7-.1-14.1 1.4-14.1h.1c22.5 2 34.3 23.8 34.3 23.8 11.2 19.6 26.2 25.1 39.6 25.1a63 63 0 0025.6-6c2-14.8 7.8-24.9 14.2-30.7-49.7-5.8-102-25.5-102-113.5 0-25.1 8.7-45.6 23-61.6-2.3-5.8-10-29.2 2.2-60.8a18.64 18.64 0 015-.5c8.1 0 26.4 3.1 56.6 24.1a208.21 208.21 0 01112.2 0c30.2-21 48.5-24.1 56.6-24.1a18.64 18.64 0 015 .5c12.2 31.6 4.5 55 2.2 60.8 14.3 16.1 23 36.6 23 61.6 0 88.2-52.4 107.6-102.3 113.3 8 7.1 15.2 21.1 15.2 42.5 0 30.7-.3 55.5-.3 63 0 5.4 3.1 11.5 11.4 11.5a19.35 19.35 0 004-.4C415.9 449.2 480 363.1 480 261.7 480 134.9 379.7 32 256 32z"
                />
              </svg>
            </span>
          </div>
          <span
            class="c17"
          >
            Change icon
          </span>
        </div>
        <span
          class="c18"
        >
          Delete
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
